spec/controllers/application_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe ApplicationController, type: :controller do
+  login
+  let(:user) { controller.current_user }
+
+  describe "#current_order" do
+    describe "current user present" do
+      it "returns the latest order of current user" do
+        FactoryBot.create(:order, user: user)
+        current_order = user.orders.last
+        expect(current_order).to be_present
+      end
+    end
+
+    describe "current user not present" do
```

Comment:
こちらも同じですね。
一応英語で書いておきます！

Well, for this rspec, I think the case you wanted to test is "when current_user doesn't exist (= user hasn't logged in), new order is created".
However, since you have called `login` at the beginning of rspec, I think `current_user` exists and what you are actually testing is the case "when current_user exists but does not have any order associated with it"

So, I guess what you should fix depends on which case you intended to test.

If this rspec was intended to test the former case (`current_user` doesn't exist), it should
* NOT call `login`
* call `controller.current_order` and check that a new order (with persisted? == false) is returned

However, if this rspec was intended to test the latter (`current_user` exists but an order doesn't exist for that user), it should
* call `login`
* (don't create order via factory)
* call `controller.current_order` and check that a new order (with persisted? == false) associated with current_user is returned

In either case, I think this rspec is not testing what is intended :(

Reply:
> If this rspec was intended to test the former case (`current_user` doesn't exist), it should
> 
> * NOT call `login`
> * call `controller.current_order` and check that a new order (with persisted? == false) is returned

This is what I intended to do, thanks for pointing that out. I also updated new code to fix these issues. Please check It when you have time 🙇